packages/hooks/src/useOauth2.ts

```diff
@@ -0,0 +1,33 @@
+import { useIsOAuth2Enabled, TOAuth2EnabledAppList, useOAuth2 } from '@deriv-com/auth-client';
+import useGrowthbookGetFeatureValue from './useGrowthbookGetFeatureValue';
+
+/**
+ * Provides an object with two properties: `isOAuth2Enabled` and `oAuthLogout`.
+ *
+ * `isOAuth2Enabled` is a boolean that indicates whether OAuth2 is enabled.
+ *
+ * `oAuthLogout` is a function that logs out the user of the OAuth2-enabled app.
+ *
+ * The `handleLogout` argument is an optional function that will be called after logging out the user.
+ * If `handleLogout` is not provided, the function will resolve immediately.
+ *
+ * @param {{ handleLogout?: () => Promise<void> }} [options] - An object with an optional `handleLogout` property.
+ * @returns {{ isOAuth2Enabled: boolean; oAuthLogout: () => Promise<void> }}
+ */
+const useOauth2 = ({ handleLogout }: { handleLogout?: () => Promise<void> } = {}) => {
```

Comment:
[Q] do we have a case where we don't pass handleLogout?

Reply:
Yes, im passing down isOauth2Enabled flag. So ill be using the hook just to check isOauth enabled or not. No need to pass logout function